xds/src/test/java/io/grpc/xds/LookasideChannelLbTest.java

```diff
@@ -0,0 +1,433 @@
+                    .setNumerator(3)
+                    .build())
+                .build())
+
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_2").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.TEN_THOUSAND)
+                    .setNumerator(45)
+                    .build())
+                .build())
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_3").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.MILLION)
+                    .setNumerator(6789)
+                    .build())
+                .build())
+            .build())
+        .build();
+    serverResponseWriter.onNext(
+        DiscoveryResponse.newBuilder()
+            .addResources(Any.pack(clusterLoadAssignment))
+            .setTypeUrl("type.googleapis.com/envoy.api.v2.ClusterLoadAssignment")
+            .build());
+
+    verify(adsStreamCallback, never()).onAllDrop();
+    verify(localityStore).updateDropPercentage(ImmutableList.of(
+        new DropOverload("cat_1", 300_00),
+        new DropOverload("cat_2", 45_00),
+        new DropOverload("cat_3", 6789)));
+
+
+    clusterLoadAssignment = ClusterLoadAssignment.newBuilder()
+        .setPolicy(Policy.newBuilder()
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_1").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.HUNDRED)
+                    .setNumerator(3)
+                    .build())
+                .build())
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_2").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.HUNDRED)
+                    .setNumerator(101)
+                    .build())
+                .build())
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_3").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.HUNDRED)
+                    .setNumerator(23)
+                    .build())
+                .build())
+            .build())
+        .build();
+    serverResponseWriter.onNext(
+        DiscoveryResponse.newBuilder()
+            .addResources(Any.pack(clusterLoadAssignment))
+            .setTypeUrl("type.googleapis.com/envoy.api.v2.ClusterLoadAssignment")
+            .build());
+
+    verify(adsStreamCallback).onAllDrop();
+    verify(localityStore).updateDropPercentage(ImmutableList.of(
+        new DropOverload("cat_1", 300_00),
+        new DropOverload("cat_2", 100_00_00)));
+
+    verify(adsStreamCallback, never()).onError();
+
+    lookasideChannelLb.shutdown();
+  }
+
+  @Test
+  public void verifyUpdateLocalityStore() {
```

Comment:
May rename to `handleLocalityAssignmentUpdates`.

Reply:
Done. I'm not a fan of the new name either though.